tests/gem5/stats-reset/configs/x86-ubuntu-run-end-reset.py

```diff
@@ -0,0 +1,196 @@
+
+memory = DualChannelDDR4_2400(size="3GiB")
+
+processor = SimpleSwitchableProcessor(
+    starting_core_type=CPUTypes.ATOMIC,
+    switch_core_type=CPUTypes.TIMING,
+    isa=ISA.X86,
+    num_cores=2,
+)
+
+board = X86Board(
+    clk_freq="3GHz",
+    processor=processor,
+    memory=memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+workload = obtain_resource("x86-ubuntu-24.04-boot-with-systemd")
+board.set_workload(workload)
+
+
+def max_tick_handler_end_reset():
+    m5.stats.reset()
+    m5.stats.dump()
+    yield True
+
+
+simulator = Simulator(
+    board=board,
+    on_exit_event={
+        ExitEvent.MAX_TICK: max_tick_handler_end_reset(),
+    },
+)
+
+simulator.set_max_ticks(15_000_000_000)  # 15 ms
+simulator.run()
+
+end_reset_dict = {}
+no_reset_dict = {}
+
+
+def read_stats_files(filepath: str, stats_dict: dict) -> None:
+    with open(filepath) as stats:
+        for line in stats:
+            tmp = line.split()
+            if len(tmp) > 1:
+                stats_dict[tmp[0]] = tmp[1]
+    stats_dict.pop("----------", None)
+
+
+read_stats_files("./gem5/stats-reset/base-case-stats.txt", no_reset_dict)
+read_stats_files(f"{m5.options.outdir}/stats.txt", end_reset_dict)
+
+# These stats are either constant, should carry over across resets, or have to
+# do with the host
+exclude_from_check = [
+    "finalTick",
+    "simFreq",
+    "hostSeconds",
+    "hostMemory",
+    "UNDEFINED",
+    "clk_domain.clock",
+    "voltage_domain.voltage",
+    "peakBW",
+]
+
+# These are the stats that don't reset when m5.stats.reset() is called, and
+# which I don't know if they are supposed to reset
+# idleFraction is in the uncertain list because it could be calculated from the
+# total number of ticks instead of the number of ticks elapsed between resets.
+uncertain_exclude_from_check = [
+    "tagsInUse",
+    "occupancies",
+    "avgOccs",
+    "ageTaskId_1024",
+    "ratioOccsTaskId",
+    "pwrStateTime::IDLE",
+    "pwrStateResidencyTicks::ON",
+    "pwrStateResidencyTicks::OFF",
+    "idleFraction",
+    "notIdleFraction",
+]
+
+
+# checks to see if any items in a list match with a key/ part of a key.
+# This is used to filter for stats that appear for several components
+def is_match_key(key: str, match_list: list[str]) -> bool:
+    if len([item for item in match_list if item in key]) == 0:
+        return False
+    else:
+        return True
+
+
+not_reset_properly = {}
+missing_keys = {}
+
+# If a stat is present when you reset but isn't there if you don't reset,
+# something is wrong
+for key, value in end_reset_dict.items():
+    if key not in no_reset_dict and not (
+        math.isnan(float(value)) or float(value) == 0.0
+    ):
+        # commented out to let tests pass
+        # not_reset_properly[key] = [value, "missing from no_reset_dict"]
+        missing_keys[key] = [value, "missing from no_reset_dict"]
+
+print("Missing keys:")
+for item in missing_keys.items():
+    print(item)
```

Comment:
Why are we storing this "missing from no_reset_dict" string in the `missing_keys` dictionary? Doesn't the fact a key is in the missing key output imply it's missing?

Why not just have `missing_keys[key] = value`?

The when you print I think you could just do:

```py
print("Missing keys")
for key in missing_keys:
    print(key)
```